### PR TITLE
Nav tweaks

### PR DESCRIPTION
**Pull request by mxsnyder**
You only need to look at the last 4 commits here. This is dependent on the **locations-page** branch.

### CODE REVIEW

File: js/page-specific/branches-filter.js [hunk 1]
```diff
@@ -71,37 +77,23 @@ bcpl.pageSpecific.filter = (($) => {
 		console.log('err', errorThrown);
```
Comment (martypowell) [inline js/page-specific/branches-filter.js]: Should this be gone?

File: js/page-specific/branches-filter.js [hunk 2]
```diff
@@ -71,37 +77,23 @@ bcpl.pageSpecific.filter = (($) => {
...
-
-		if ($target.parent().is('#sort-control')) {
-			$branches.fadeOut(250, () => {
-				const $branchCards = $branches.find('.card').detach();
-				$branches.append($branchCards.get().reverse());
-				$branches.fadeIn(250);
-			});
-		}
+	const amenitiesShowing = (collapseEvent) => {
+		$(collapseEvent.currentTarget).siblings('.collapse-control').html('<i class="fa fa-minus"></i> Hide Filters');
```
Comment (martypowell) [inline js/page-specific/branches-filter.js]: new lines for .siblings and .html

File: js/page-specific/branches-filter.js [hunk 3]
```diff
@@ -71,37 +77,23 @@ bcpl.pageSpecific.filter = (($) => {
...
+	};
 
-		if ($target.parent().is('#list-grid-control')) {
-			$branches.fadeOut(250, () => {
-				$branches.toggleClass('list-view');
-				$branches.fadeIn(250);
-			});
-		}
+	const amenitiesHiding = (collapseEvent) => {
+		$(collapseEvent.currentTarget).siblings('.collapse-control').html('<i class="fa fa-plus"></i> Show Filters');
```
Comment (martypowell) [inline js/page-specific/branches-filter.js]: new lines for .siblings and .html

File: js/page-specific/branches-map.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+	let infowindows = [];
+
+	const getAddressForDirections = (branch) => {
+		return [branch.address, branch.city, 'MD', branch.zip].join('+').replace(' ', '+');
+	};
+
+	const processBranchData = (branchData) => {
+		const branchJson = typeof branchData === 'string' ? JSON.parse(branchData) : branchData;
+
+		branchJson.forEach((branch) => {
```
Comment (martypowell) [inline js/page-specific/branches-map.js]: should this be in it's own function?

File: js/page-specific/branches-map.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+				marker.addListener('click', () => {
+					infowindow.open(map, marker);
+				});
+			}
+		});
+	};
+
+	const reportBranchDataError = (err) => {
+		console.log(err);
```
Comment (martypowell) [inline js/page-specific/branches-map.js]: good for prod?

File: js/window-shade.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+namespacer('bcpl');
+
+bcpl.windowShade = (($) => {
+	const windowShadeSelector = '#window-shade';
+	let timeout;
+
+	const cycle = (displaySpeed, delaySpeed) => {
+		const $windowShade = $(windowShadeSelector);
+		clearTimeout(timeout);
+		$windowShade.slideDown(displaySpeed, () => {
```
Comment (martypowell) [inline js/window-shade.js]: move to it's own function?

File: test/navigation.spec.js [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+			loadFixtures('menuAndModal.fixture.html');
+			done();
+		});
+
+		it('should focus the first active sliding navigation menu link', () => {
+			$('#responsive-sliding-navigation li').first().addClass('active');
+
+			bcpl.navigation.focusFirstActiveMenuLink();
+
+			expect($('#responsive-sliding-navigation li.active a').first().is(':focus')).toBe(true);
```
Comment (martypowell) [inline test/navigation.spec.js]: General comment, i typically would store ```javascript $('#responsive-sliding-navigation li.active a').first().is(':focus')``` in a variable...enhances readability...what do you think?

File: test/navigation.spec.js [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+			const $randomLink = $('nav li li a').eq(randomLinkIndex);
+			const $actualButton = $randomLink.closest('nav>ul>li').find('button');
+
+			const $experimental = bcpl.navigation.findClosestButtonToLink($randomLink);
+
+			expect($experimental.is($actualButton)).toBe(true);
+		});
+	});
+
+	describe('afterSubmenuActivated', () => {});
```
Comment (martypowell) [inline test/navigation.spec.js]: assuming this are TODO?

File: test/spec-runner.html [hunk 9]
```diff
@@ -19,10 +19,15 @@
 
 		<!-- JavaScripts -->
 		<script src="/js/navigation-search.js"></script>
-
+		<script src="/js/alert-box.js"></script>
+		<script src="/js/utility/namespacer.js"></script>
+		<script src="/js/alert-box.js"></script>
```
Comment (martypowell) [inline test/spec-runner.html]: Uneeded reference, you referenced this in line 22.

File: js/page-specific/branches-filter.js (hunk 2 above)
Comment (mxsnyder) [reply js/page-specific/branches-filter.js]: Done

File: js/page-specific/branches-filter.js (hunk 3 above)
Comment (mxsnyder) [reply js/page-specific/branches-filter.js]: Done

File: js/page-specific/branches-filter.js (hunk 1 above)
Comment (mxsnyder) [reply js/page-specific/branches-filter.js]: We need a good idea for client-side error handling first.

File: js/page-specific/branches-map.js (hunk 5 above)
Comment (mxsnyder) [reply js/page-specific/branches-map.js]: We need a good idea for client-side error handling first.

File: js/page-specific/branches-map.js (hunk 4 above)
Comment (mxsnyder) [reply js/page-specific/branches-map.js]: Done

File: js/window-shade.js (hunk 6 above)
Comment (mxsnyder) [reply js/window-shade.js]: The problem is that I'd have to send params into setTimeout beyond the function and the expiration, which isn't supported in IE9 (we still support it).

File: test/navigation.spec.js (hunk 7 above)
Comment (mxsnyder) [reply test/navigation.spec.js]: Done

File: test/navigation.spec.js (hunk 8 above)
Comment (mxsnyder) [reply test/navigation.spec.js]: Correct. I stubbed out the whole object, then had to switch gears after the first 4 or 5.

File: test/spec-runner.html (hunk 9 above)
Comment (mxsnyder) [reply test/spec-runner.html]: Done!

**Pull request merged**